### DISCUSSION TITLE
Renamed apk is not visible as artifacts in slack notification for download

### DISCUSSION THREAD

**Discussion by hashirshoaeb** [codemagic-ci-cd/codemagic-docs#2155] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

I'm using workflow editor and added a post-build script, to rename the apk file and added it in artifacts but it is not visible in slack notification.

here is my script: 
```bash
#!/bin/sh
set -e
set -x

echo "renaming apk file to $CM_TAG";

if [ -n "$CM_TAG" ]; then
  mv build/app/outputs/flutter-apk/app-qa-release.apk build/app/outputs/flutter-apk/$CM_TAG.apk;
  cp build/app/outputs/flutter-apk/$CM_TAG.apk "$CM_EXPORT_DIR";
fi

```

### Expected results

Notification should have link for apk under artifacts

#### slack notificaiton without script
![Screenshot 2023-10-27 at 9 44 22 AM](https://github.com/codemagic-ci-cd/codemagic-docs/assets/35165481/b7fa6559-5dd6-44d8-aeab-3c4af1f9e887)
apk is visible under artifacts

#### notification after renaming the file using script
![Screenshot 2023-10-27 at 9 46 12 AM](https://github.com/codemagic-ci-cd/codemagic-docs/assets/35165481/f9277b80-d6d0-48a1-8e52-2468953239b9)

**Comment by dtrdic** [comment, marked as answer]
Hey @hashirshoaeb, unfortunately this won't be possible with the `Workflow editor`. 
The reason is that artifacts are collected **right after** the build steps are completed, in comparison to `YAML` where artifacts are collected **before publishing** them so there is a chance to run a pre-publish script. However, with the Workflow editor, the sequence of steps doesn't allow for such pre-publish scripting, which is why you're not seeing the desired results in the Slack notification
